config/src/lib.rs

```diff
@@ -108,8 +108,10 @@ impl ClientConfigBuilder {
             if !enable {
                 return vec![];
             }
-        } else {
+        } else if !self.default_is_tls_enabled() {
             return vec![];
+        } else {
```

Comment:
Else is not needed

Reply:
I agree with you, but clippy "else_if_without_else" disagrees with both of us.

I applied the workaround recommended at https://github.com/rust-lang/rust-clippy/blob/master/clippy_lints/src/else_if_without_else.rs